packages/service-proxy/src/__tests__/unit/mixin/service.mixin.unit.ts

```diff
@@ -78,7 +78,7 @@ describe('ServiceMixin', () => {
     const boundServices = myApp.find('services.*').map(b => b.key);
     expect(boundServices).to.containEql('services.GeocoderService');
     const binding = myApp.getBinding('services.GeocoderService');
-    expect(binding.scope).to.equal(BindingScope.TRANSIENT);
+    expect(binding.scope).to.be.undefined();
```

Comment:
This looks like a backwards-incompatible change to me.

I think we should use a different approach:

- Keep the change setting the value of`binding._scope` to `undefined`
- Modify `binding.scope` getter to convert `undefined to `TRANSIENT`
- Add a new public method for applying the default scope, something along the following lines.

```ts
class Binding {
 // ..

  applyDefaultScope(scope: BindingScope) {
    if (this._scope) return;
    this._scope = scope;
  }
}
```

That will give use two Binding methods:
 - `binding.inScope` forces the scope the the value provided
 - `binding.applyDefaultScope` preserves explicitly configured scope and applies the given scope only when no explicit scope was defined

Reply:
Good idea.